backend/sportsmatch/src/main/java/com/sportsmatch/auth/SecurityConfig.java

```diff
@@ -8,39 +10,44 @@
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.annotation.web.configurers.HeadersConfigurer;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
-
-import static org.springframework.boot.autoconfigure.security.servlet.PathRequest.toH2Console;
-import static org.springframework.security.config.http.SessionCreationPolicy.STATELESS;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
 
 @Configuration
 @EnableWebSecurity
 @RequiredArgsConstructor
 public class SecurityConfig {
 
-    private static final String[] WHITE_LIST_URL = {
-            "/api/v1/auth/**",
-            "/h2-console/**",
-            "/v3/api-docs",
-            "/v3/api-docs/**",
-            "/swagger-ui/**"
-            // add endpoints that are not authenticated
-            };
-    private final JwtAuthFilter jwtAuthFilter;
-    private final AuthenticationProvider authenticationProvider;
-    @Bean
-    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception { // set which endpoints are authenticated and not
-        http
-                .csrf(AbstractHttpConfigurer::disable)
-                .headers(h -> h.frameOptions(HeadersConfigurer.FrameOptionsConfig::disable)) // remove in production
-                .authorizeHttpRequests(r -> r
-                        .requestMatchers(WHITE_LIST_URL)
-                        .permitAll()
-                        .anyRequest().authenticated())
-                .sessionManagement(s -> s.sessionCreationPolicy(STATELESS))
-                .authenticationProvider(authenticationProvider)
-                .addFilterBefore(jwtAuthFilter, UsernamePasswordAuthenticationFilter.class);
-        return http.build();
-    }
+  private static final String[] WHITE_LIST_URL = {
+    "/api/v1/auth/**", "/h2-console/**", "/v3/api-docs", "/v3/api-docs/**", "/swagger-ui/**"
+  };
+  private final JwtAuthFilter jwtAuthFilter;
+  private final AuthenticationProvider authenticationProvider;
+  private final LogoutHandler logoutHandler;
+
+  /**
+   * Configures and returns a SecurityFilterChain for HTTP security.
+   *
+   * @param http The HttpSecurity instance to configure.
+   * @return SecurityFilterChain instance.
+   * @throws Exception If an exception occurs during configuration.
```

Comment:
this is useless information. In my opinion, only add java docs when it makes sense to you, don't force yourself into it